include/ncengine/module/Module.h

```diff
@@ -19,8 +23,11 @@ struct Module
     /** @brief Called on registered modules when the task graph is constructed. */
     virtual void OnBuildTaskGraph(task::TaskGraph&) {}
 
-    /** @brief Called prior to clearing the module's associated data registry. This includes
-     *         scene changes and NcEngine::Shutdown(). */
+    /** @brief Called on registered modules prior to loading a new scene. */
+    virtual void OnSceneLoad() {}
```

Comment:
Should this be: 
`virtual void OnBeforeSceneLoad()`

The pattern already in place here with `OnBuildTaskGraph` is that "OnX" would run **after** X, unless I'm misreading the task graph comment.